subsys/net/lib/openthread/platform/radio.c

```diff
@@ -1223,3 +1252,23 @@ otError otPlatRadioConfigureEnhAckProbing(otInstance *aInstance, otLinkMetrics a
 }
 
 #endif /* CONFIG_OPENTHREAD_LINK_METRICS_SUBJECT */
+
+otError otPlatRadioSetChannelMaxTransmitPower(otInstance *aInstance, uint8_t aChannel,
+					      int8_t aMaxPower)
+{
+	ARG_UNUSED(aInstance);
+	otError error = OT_ERROR_NONE;
+
+	if (aChannel >= OT_RADIO_2P4GHZ_OQPSK_CHANNEL_MIN &&
```

Comment:
Is there a use case for invalid channel or could we switch to assert for these verification?

Reply:
Asserts can be turned off and by that It would be possible to call this function and cause memory errors. On the other hand there is no place in code that is is called with incorrect value.

Anyway I prefer to leave it as it is.

Reply:
Switching to assert here and above will save a few bytes of memory for the production builds. Any wrong usage of the function will be catched by the CI which will always enable asserts.

Reply:
The function is used in RCP so saving few bytes doesn't matter there so much.

> Any wrong usage of the function will be catched by the CI which will always enable asserts.

Since this is RCP the function that calls it do not have to be in the same host even.

So still I prefer to leave it as it is.

Reply:
Okay, accepting since both simulation and posix implementations have similar check.